build.gradle

```
@@ -503,6 +507,10 @@ subprojects {
     if (shouldUseJUnit5) {
       useJUnitPlatform {
         excludeTags "integration"
+        // KAFKA-14109
+        // Both engines are needed to run JUnit 4 tests alongside JUnit 5 tests.
+        // junit-vintage (JUnit 4) can be removed once the JUnit 4 migration is complete.
+        includeEngines "junit-vintage", "junit-jupiter"
       }
```

Here we can do similar as above but excluding the tags.
```suggestion
      if (project.name.equals('streams')) {
        useJUnitPlatform {
          excludeTags "integration"
          excludeTags 'org.apache.kafka.test.IntegrationTest'
          includeEngines "junit-vintage", "junit-jupiter"
        }
      } else {
        useJUnitPlatform {
          excludeTags "integration"
        }
      }
```